contracts/RealmDiamond/facets/ParcelRolesRegistryFacet.sol

```diff
@@ -208,6 +254,60 @@ contract ParcelRolesRegistryFacet is Modifiers, IERC7432 {
     revert("ParcelRolesRegistryFacet: sender is not approved");
   }
 
+  /// @notice Decodes the profit share data from the provided calldata.
+  /// @param data The encoded profit share data.
+  /// @return tokenAddresses The array of token addresses involved in the profit share.
+  /// @return ownerShares The array of token addresses involved in the profit share. // CHANGE THE TEXT
+  /// @return borrowerShares The array of token addresses involved in the profit share.
+  /// @return shares The 2D array representing shares of each recipient.
+  /// @return recipients The 2D array representing the recipient addresses for each token.
+  function _decodeProfitShare(
+    bytes calldata data
+  )
+    internal
+    pure
+    returns (
+      address[] memory tokenAddresses,
+      uint16[] memory ownerShares,
+      uint16[] memory borrowerShares,
+      uint16[][] memory shares,
+      address[][] memory recipients
+    )
+  {
+    return abi.decode(data, (address[], uint16[], uint16[], uint16[][], address[][]));
+  }
+
+  function _handleProfitShareData(
+    address _tokenAddress,
+    uint256 _tokenId,
+    bytes32 _roleId,
+    address[] memory profitTokens,
+    uint16[] memory ownerShares,
+    uint16[] memory borrowerShares,
+    uint16[][] memory sharesArray,
+    address[][] memory recipientsArray
+  ) internal {
+    s.profitShares[_tokenAddress][_tokenId][_roleId] = ProfitShare(ownerShares, borrowerShares, profitTokens, sharesArray, recipientsArray);
+
+    for (uint256 i = 0; i < profitTokens.length; i++) {
+      _validateShares(sharesArray[i], ownerShares[i], borrowerShares[i]);
+    }
+  }
+
+  /// @notice Validates the shares for a single token.
+  /// @param shares The shares array for the token.
+  /// @param ownerShare The global owner share.
+  /// @param borrowerShare The global borrower share.a
+  function _validateShares(uint16[] memory shares, uint16 ownerShare, uint16 borrowerShare) internal pure {
+    require(shares.length >= 0, "ParcelRolesRegistryFacet: Each shares array must have at least owner and borrower shares");
```

Comment:
This is not true, if the profitShare only has the owner and the borrower shares, the shares array will be empty

Reply:
true